src/operations/create_collection.ts

```diff
@@ -55,16 +50,23 @@ class CreateCollectionOperation extends CommandOperation {
       }
     }
 
-    const cmd: any = { create: name };
-    for (let n in options) {
-      if (
-        options[n] != null &&
-        typeof options[n] !== 'function' &&
-        !ILLEGAL_COMMAND_FIELDS.has(n)
-      ) {
-        cmd[n] = options[n];
-      }
-    }
+    const cmd = {
+      create: name,
+      ...options
+    };
+    delete cmd.w;
```

Comment:
Let's try to never use the `delete` keyword. Instead I think we need a TypeScript utility that let's us pluck a list of named options and apply them to a `Document`